src/main/java/com/mamu/todo_app/mapper/TodoMapper.java

```diff
@@ -0,0 +1,56 @@
+package com.mamu.todo_app.mapper;
+
+import com.mamu.todo_app.dto.TodoDTO;
+import com.mamu.todo_app.model.Todo;
+import com.mamu.todo_app.types.StatusType;
+import org.springframework.stereotype.Component;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.Date;
+
+@Component
+public class TodoMapper {
+    public TodoDTO toTodoDTO(Todo todo){
+        return TodoDTO.builder()
+                .title(todo.getTitle())
+                .description(todo.getDescription())
+                .status(getStatusString(todo.getStatus()))
+                .targetDate(todo.getTargetDate().toString())
+                .build();
+    }
+
+    public Todo toTodo (TodoDTO todoDTO) throws ParseException {
+        return Todo.builder()
+                .title(todoDTO.getTitle())
+                .description(todoDTO.getDescription())
+                .status(getStatusType(todoDTO.getStatus()))
+                .targetDate(getDate(todoDTO.getTargetDate()))
+                .build();
+    }
+    public String getStatusString (StatusType statusType){
```

Comment:
It is not maintainable if you need to add more status you need to add more switch cases =, maybe you could try to use enums with constructors and add some methods to return the string value, or return and enum from a string value

Reply:
The problem has been fixed :)